src/main/java/frc/robot/subsystems/Elevator/ElevatorSubsystem.java

```diff
@@ -0,0 +1,156 @@
+    private final XboxController mechController; 
+
+    public ElevatorSubsystem(){
+        //Print out current position for debug & measurement
+        //System.out.print(extensionEncoder.getPosition());
+        
+        elevatorNetworkTableInstance = NetworkTableInstance.getDefault();
+        elevatorNetworkTable = elevatorNetworkTableInstance.getTable("elevator");
+        elevatorNetworkTable.addListener("target_position", EnumSet.of(NetworkTableEvent.Kind.kValueAll), this::acceptNewPosition);
+
+        extensionMotor = new CANSparkMax(Constants.ElevatorConstants.EXTENSION_ID, MotorType.kBrushless);
+        extensionMotor.setIdleMode(IdleMode.kBrake);
+        
+        extensionEncoder = extensionMotor.getEncoder();
+        extensionEncoder.setPositionConversionFactor(Constants.ElevatorConstants.POSITIONCONVERSIONFACTOR);
+        extensionEncoder.setVelocityConversionFactor(Constants.ElevatorConstants.VELOCITYCONVERSIONFACTOR);
+        extensionEncoder.setPosition(0);
+        
+        extensionFollow = new CANSparkMax(Constants.ElevatorConstants.EXTENSION_FOLLOW_ID, MotorType.kBrushless);
+        extensionFollow.follow(extensionMotor);
+        extensionFollow.setIdleMode(IdleMode.kBrake);
+
+        extensionPidController = extensionMotor.getPIDController();
+        extensionPidController.setP(extensionP);
+        extensionPidController.setI(extensionI);
+        extensionPidController.setD(extensionD);
+        extensionPidController.setSmartMotionAllowedClosedLoopError(extensionTolerance, 0);
+        
+        zeroLimitSwitch = new DigitalInput(Constants.ElevatorConstants.ZERO_LIMIT_ID);
+
+        //Controller for testing.
+        mechController = new XboxController(Constants.OperatorConstants.kDriverControllerPort);
+    }
+    @Override
+    public void periodic(){
+        
+        if(IS_MANUAL){
+            //Add some factors for better control.
+            extensionMotor.set(mechController.getRightY());
+            return;
+        }
+        if (zeroLimitSwitch != null && !zeroLimitSwitch.get()){
+            extensionEncoder.setPosition(0); 
+        }
+
+        extensionMotor.enableSoftLimit(SoftLimitDirection.kReverse, true);
+
+        //Start move to target posision
+        if (targetState != state){
+            extensionPidController.setReference(targetState.getExtension(), ControlType.kPosition, 0, 0.03, ArbFFUnits.kPercentOut);
+        }
+
+        if(mechController.getAButtonPressed()){
+            this.setTargetState(ElevatorState.GROUND);
+        }
+        else if(mechController.getBButtonPressed()){
+            this.setTargetState(ElevatorState.AMP);
+        }
+        else if(mechController.getXButtonPressed()){
+            this.setTargetState(ElevatorState.SPEAKER);
+        }
+        else if(mechController.getYButtonPressed()){
+            this.setTargetState(ElevatorState.CHUTE);
+        }
+        
+    }
+    
+    public void setState(ElevatorState state) {
```

Comment:
never used -> get state will not be correct -> all of the go to position commands will not work properly. I recommend having a function of atState() which returns if the current extension is within a small tolerance of the set state, and use this function for the commands.